### PR TITLE
RELATED: SD-261 - fix polling

### PR DESCRIPTION
**Pull request by ptquang86**
PR for [SD-261](https://jira.intgdc.com/browse/SD-261)
- Fix export polling
- Add getHeaders into ApiResponse
- Add handleHeadPolling into utils

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/181/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/213/ : SUCCESS in 39s

File: src/util.ts [hunk 1]
```diff
@@ -55,7 +55,13 @@ export const handlePolling = (
     } = options;
 
     return xhrRequest(uri)
-        .then((r: any) => r.getData())
+        .then((r: any) => {
+            let res = r.getData();
+            if (res === '') {
```
Comment (martinmosko) [inline src/util.ts]: this is fragile solution... :/ As you can see handlePooling is designed only to handle pooling through responseBody data... its not for handling response itself.

please do not modify handlePooling method but rather make a copy and set required handling...

you're trying to bend existing solution too much where `handlePooling` is not designed for generic solution...

or you have to completely rewrite handlePooling to handle different xhrRequests properly

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/187/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/219/ : SUCCESS in 48s

**Comment by ptquang86** [comment]
Thanks @martinmosko, please review again :)

**Comment by martinmosko** [comment]
@ptquang86 please fix ticket number in commit message :)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/188/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/220/ : SUCCESS in 36s

**Comment by ptquang86** [comment]
@martinmosko, the commit message has been updated, thanks

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/52/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/223/ : FAILURE in 40s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/53/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/224/ : SUCCESS in 55s

**Pull request merged**